### PR TITLE
Compute v2: Delete aggregate

### PR DESCRIPTION
**Pull request by dstdfx**
For #738 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

API doc: 
- https://developer.openstack.org/api-ref/compute/#delete-aggregate

Source code: 
- https://github.com/openstack/nova/blob/master/nova/api/openstack/compute/aggregates.py#L124

### CODE REVIEW

**Comment by jtopjian** [comment]
@dstdfx This will need rebased. Looks good to me, otherwise.

**Comment by dstdfx** [comment]
@jtopjian Done :)

**Comment by jtopjian** [comment]
@dstdfx I did a quick test of this and it looks like Nova is returning a 200 code. Can you confirm if that's what you're seeing?

```
=== RUN   TestAggregatesDelete
--- FAIL: TestAggregatesDelete (3.10s)
        aggregates_test.go:21: Expected HTTP response code [202 204] when accessing [DELETE http://localhost:8774/v2.1/6c3c1bb5e47a4c5982b8a87f61c62027/os-aggregates/1], but got 200 instead
```

**Comment by dstdfx** [comment]
@jtopjian You're right. I've changed it.

File: openstack/compute/v2/extensions/aggregates/requests.go [hunk 1]
```diff
@@ -39,3 +41,11 @@ func Create(client *gophercloud.ServiceClient, opts CreateOpts) (r CreateResult)
 	})
 	return
 }
+
+// Delete makes a request against the API to delete an aggregate.
+func Delete(client *gophercloud.ServiceClient, aggregateID int) (r DeleteResult) {
+	v := strconv.Itoa(aggregateID)
+	_, r.Err = client.Delete(aggregatesDeleteURL(client, v), &gophercloud.RequestOpts{
+		OkCodes: []int{200}})
```
Comment (jtopjian) [inline openstack/compute/v2/extensions/aggregates/requests.go]: This is a very minor nit and I apologize, but we should make this look like the other OkCodes:

```go
_, r.Err = client.Delete(aggregatesDeleteURL(client, v), &gophercloud.RequestOpts{
        OkCodes: []int{200},
})

Comment (dstdfx) [reply openstack/compute/v2/extensions/aggregates/requests.go]: Wow, I'm surprised that `go fmt` didn't fix it.

Comment (jtopjian) [reply openstack/compute/v2/extensions/aggregates/requests.go]: Yeah, I was surprised that both `fmt` _and_ `vet` didn't flag it. Thank you for fixing this :)

**Comment by jtopjian** [comment]
@dstdfx Thank you for your work on this. If you can, can you submit a PR with an acceptance test for this?

**Comment by dstdfx** [comment]
@jtopjian Sure, next PR will be an acceptance test for create/delete aggregates.

**Pull request merged**